stable_diffusion_ov/openvino_models/model_collateral_cache.cpp

```diff
@@ -65,51 +68,72 @@ namespace cpp_stable_diffusion_ov
         std::string vae_decoder_model_name = model_folder + OS_SEP + "vae_decoder.xml";
         std::string vae_encoder_model_name = model_folder + OS_SEP + "vae_encoder.xml";
 
-
-        if ((model_folder.find("int8") != std::string::npos) || (model_folder.find("INT8") != std::string::npos))
+        //if the unet loop isn't set, or it's set but either unet+ / unet- devices have changed.
+        if (!_unet_loop || !_unet_positive_device || !_unet_negative_device || (*_unet_positive_device != unet_positive_device) ||
+            (*_unet_negative_device != unet_negative_device))
         {
-            if (!_unet_loop || !_unet_positive_device || !_unet_negative_device || (*_unet_positive_device != unet_positive_device) ||
-                (*_unet_negative_device != unet_negative_device))
+            std::string unet_model_folder = model_folder;
+            if (unet_sub_dir)
+            {
+                unet_model_folder = unet_model_folder + OS_SEP + *unet_sub_dir;
+            }
+
+            if ((unet_model_folder.find("int8") != std::string::npos) || (unet_model_folder.find("INT8") != std::string::npos))
             {
                 std::cout << "Creating new int8 unet loop with +/- devices as: " << unet_positive_device << ", " << unet_negative_device << std::endl;
-                _unet_loop = std::make_shared< UNetLoopSD15InternalBlobs >(model_folder,
+                _unet_loop = std::make_shared< UNetLoopSD15InternalBlobs >(unet_model_folder,
                     unet_positive_device,
                     unet_negative_device,
                     tok_max_length,
                     512,
```

Comment:
Does hardcoding dimensions here (and below) prevent the usage of portrait or landscape models? would be great to get these dims from the model instead of forcing it to only 512x512